### PR TITLE
Add line item granularity

### PR DESCRIPTION
**Pull request by smcjones**
- Added line items to granularity functions
- Changed default behavior to throw an error
- Added unit tests

### CODE REVIEW

File: ts/dv360/src/types.ts [hunk 1]
```diff
@@ -173,3 +177,22 @@ export interface Accessors {
   insertionOrders: typeof InsertionOrders;
   lineItems: typeof LineItems;
 }
+
+/**
+ * A map of an ID to a resource.
+ */
+export interface EntityMap<Resource extends DisplayVideoResource> {
+  [entityId: string]: Resource;
```
Comment (datumest) [inline ts/dv360/src/types.ts:185]: I know we've discussed this before and we're long past the use of this pattern here, but I would like to still point out [this section](https://google.github.io/styleguide/tsguide.html#mapped-conditional-types:~:text=type%20operators%20require%20the%20reader%20to%20mentally%20evaluate%20the%20type%20expression.%20This%20can%20make%20programs%20substantially%20harder%20to%20read) of the style guide, which I feel is still applicable here. Object.values() feels like a code smell, and is obfuscates understanding the type. But again, we are well past that here, so I won't let that be a blocker.

**Review by datumest — APPROVED** [review]

**Review by anaesqueda — APPROVED** [review]
LGTM

File: ts/dv360/src/types.ts (hunk 1 above)
Comment (smcjones) [reply ts/dv360/src/types.ts]: No, you're right. It's better to clean up if we see it too.

Is this OK? It really doesn't show up much.

File: ts/dv360/src/client.ts [hunk 2]
```diff
@@ -195,44 +195,60 @@ export class Client implements ClientInterface {
...
-              (arr, { advertiserId }) =>
-                arr.concat(
-                  this.getAllInsertionOrdersForAdvertiser(advertiserId),
-                ),
-              [] as InsertionOrder[],
-            );
+
+  getAllInsertionOrders(): { [id: string]: InsertionOrder } {
+    function entries(io: InsertionOrder) {
+      return [io.getId(), io] satisfies [id: string, resource: InsertionOrder];
```
Comment (smcjones) [inline ts/dv360/src/client.ts:222]: @datumest shows the changes

Comment (datumest) [reply ts/dv360/src/client.ts:222]: Cleaner, thank you!

**Pull request merged**